sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/UrlSanitizer.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.http.policy;
+
+import com.azure.core.implementation.ImplUtils;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.UrlBuilder;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.azure.core.implementation.logging.LoggingKeys.REDACTED_PLACEHOLDER;
+
+public final class UrlSanitizer {
+    static final List<String> DEFAULT_QUERY_PARAMS_ALLOWLIST = Collections.singletonList("api-version");
+    private final Set<String> allowedQueryParamNames;
+
+    public UrlSanitizer(Collection<String> allowedQueryParamNames) {
+        Collection<String> allowedParams
+            = allowedQueryParamNames == null ? DEFAULT_QUERY_PARAMS_ALLOWLIST : allowedQueryParamNames;
+        this.allowedQueryParamNames = allowedParams.stream()
+            .map(queryParamName -> queryParamName.toLowerCase(Locale.ROOT))
+            .collect(Collectors.toSet());
```

Comment:
We should have a constant Set for the default allowed query params instead of creating one each time. Or better yet, maybe we should turn this into a `Predicate<String>` as the default only allows a single value which can be optimized to `"api-version".equals(paramName)`